File: backend/src/main/java/com/example/backend/modules/hospital/entity/Hospital.java
package com.example.backend.modules.hospital.entity;

import com.example.backend.common.base.BaseEntity;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;

/**
 * 医院信息实体类
 */
@Data
@Entity
@Table(name = "t_hospital")
@EqualsAndHashCode(callSuper = true)
public class Hospital extends BaseEntity {

    /**
     * 医院名称
     */
    @Column(nullable = false, length = 100)
    private String name;

    /**
     * 医院等级
     */
    @Column(nullable = false)
    private Integer level = 1;

    /**
     * 医院类型
     */
    @Column(nullable = false)
    private Integer type = 1;

    /**
     * 省份
     */
    @Column(nullable = false, length = 50)
    private String province;

    /**
     * 城市
     */
    @Column(nullable = false, length = 50)
    private String city;

    /**
     * 区县
     */
    @Column(nullable = false, length = 50)
    private String district;

    /**
     * 详细地址
     */
    @Column(nullable = false, length = 200)
    private String address;

    /**
     * 联系电话
     */
    @Column(length = 20)
    private String phone;

    /**
     * 邮箱
     */
    @Column(length = 100)
    private String email;

    /**
     * 官网地址
     */
    @Column(length = 200)
    private String website;

    /**
     * 医院简介
     */
    @Column(columnDefinition = "text")
    private String description;

    /**
     * 医院图片(多个逗号分隔)
     */
    @Column(length = 1000)
    private String images;

    /**
     * 营业时间
     */
    @Column(length = 100)
    private String businessHours;

    /**
     * 经度
     */
    @Column(precision = 10, scale = 7)
    private BigDecimal longitude;

    /**
     * 纬度
     */
    @Column(precision = 10, scale = 7)
    private BigDecimal latitude;

    /**
     * 状态
     */
    @Column(nullable = false)
    private Integer status = 1;

    /**
     * 排序
     */
    private Integer sort = 0;
}

File: vite-project/src/types/hospital.ts
import type { PageResult, Result } from './common';

// 医院等级枚举
export enum HospitalLevel {
  LEVEL_ONE = 1,
  LEVEL_TWO = 2,
  LEVEL_THREE = 3,
}

export const HospitalLevelTextMap: Record<HospitalLevel, string> = {
  [HospitalLevel.LEVEL_ONE]: '一级医院',
  [HospitalLevel.LEVEL_TWO]: '二级医院',
  [HospitalLevel.LEVEL_THREE]: '三级医院',
};

// 医院类型枚举
export enum HospitalType {
  GENERAL = 1,
  SPECIALIZED = 2,
  TRADITIONAL_CHINESE = 3,
}

export const HospitalTypeTextMap: Record<HospitalType, string> = {
  [HospitalType.GENERAL]: '综合医院',
  [HospitalType.SPECIALIZED]: '专科医院',
  [HospitalType.TRADITIONAL_CHINESE]: '中医医院',
};

// 医院状态枚举
export enum HospitalStatus {
  DISABLED = 0,
  NORMAL = 1,
}

export const HospitalStatusTextMap: Record<HospitalStatus, string> = {
  [HospitalStatus.DISABLED]: '禁用',
  [HospitalStatus.NORMAL]: '正常',
};

// 医院信息接口
export interface Hospital {
  id: number;
  name: string;
  level: number;
  levelDesc: string;
  type: number;
  typeDesc: string;
  province: string;
  city: string;
  district: string;
  address: string;
  fullAddress: string;
  phone?: string;
  email?: string;
  website?: string;
  description?: string;
  images?: string;
  businessHours?: string;
  longitude?: number;
  latitude?: number;
  status: number;
  statusDesc: string;
  sort: number;
  createdTime: string;
  updatedTime: string;
}

// 医院查询参数
export interface HospitalQueryDTO {
  name?: string;
  province?: string;
  city?: string;
  district?: string;
  level?: HospitalLevel;
  type?: HospitalType;
  status?: HospitalStatus;
}

// 医院新增参数
export interface HospitalAddDTO {
  name: string;
  level: HospitalLevel;
  type: HospitalType;
  province: string;
  city: string;
  district: string;
  address: string;
  phone?: string;
  email?: string;
  website?: string;
  description?: string;
  images?: string;
  businessHours?: string;
  longitude?: number;
  latitude?: number;
  status?: HospitalStatus;
  sort?: number;
}

// 医院更新参数
export interface HospitalUpdateDTO extends HospitalAddDTO {
  id: number;
}

// API 响应类型
export type HospitalListResult = Result<PageResult<Hospital>>;
export type HospitalDetailResult = Result<Hospital>;
export type ProvinceListResult = Result<string[]>;
export type CityListResult = Result<string[]>;
